adding test to illustrate #183.

nickerso opened the pull request:
New test parses a model which contains a component with multiple math children,
which is expected to parse fine. Due to #183 this is not the case and one of the
math children is discarded.

Commit:
adding test to illustrate #183.

New test parses a model which contains a component with multiple math children,
which is expected to parse fine. Due to #183 this is not the case and one of the
math children is discarded.

hsorby commented on tests/parser/file_parser.cpp:
```diff
@@ -108,3 +108,24 @@ TEST(Parser, parseComplexEncapsulationModelFromFile) {
+    EXPECT_EQ(0, p.errorCount());
+
+    std::string e1 = "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n\t\t\t<apply><eq/>\n\t\t\t\t<ci>a1</ci>\n\t\t\t\t<apply><plus/>\n\t\t\t\t\t<ci>b1</ci>\n\t\t\t\t\t<ci>c1</ci>\n\t\t\t\t</apply>\n\t\t\t</apply>\n\t\t</math>";
+    std::string e2 = "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n\t\t\t<apply><eq/>\n\t\t\t\t<ci>b2</ci>\n\t\t\t\t<apply><times/>\n\t\t\t\t\t<cn xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" cellml:units=\"dimensionless\">2.0</cn>\n\t\t\t\t\t<ci>d</ci>\n\t\t\t\t</apply>\n\t\t\t</apply>\n\t\t</math>\n\t\t<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n\t\t\t<apply><eq/>\n\t\t\t ...
```
Expected strings have traditionally gone at the beginning of a test.

Commit:
tidy up code ordering.

Commit:
update component math parsing to allow for the case of multiple math children.

Need to handle the case when a component has multiple math children elements, so should be
appending math rather than setting it each time.

Commit:
Correct expected string.

Forgot that when appending the math the whitespace between math elements in the file
will be dropped from the component's math string.

hsorby commented:
This all looks fine to me but I do find it a bit overkill to provide an external file resource to illustrate the issue.  I think a much simpler test could have been provided that would trigger this issue.

agarny commented:
It all looks good to me too, but waiting for a decision to be made on the external file resource 'issue', so not merging just yet.

PS: do we really want to use `\t` in CellML files rather than a certain number of spaces? It's just that more often than not people tend to advocate the latter nowadays.

nickerso commented:
A smaller test could be written, but easier for me to copy-paste code from related work :) Personally I'm not concerned how users serialise their XML.
